### PR TITLE
fix(core): fix non-existing partition drop can suspend WAL table

### PR DESCRIPTION
**Pull request by ideoma**
Fixes https://github.com/questdb/questdb/issues/4060

### CODE REVIEW

File: core/src/test/java/io/questdb/test/griffin/wal/WalAlterTableSqlTest.java [hunk 1]
```diff
@@ -158,6 +158,30 @@ public void createWalAndDropPartition() throws Exception {
         });
     }
 
+
```
Comment (bziobrowski) [inline core/src/test/java/io/questdb/test/griffin/wal/WalAlterTableSqlTest.java:161]: Would it make sense to also test dropping non-existing partition past the active one ?

**Review by bziobrowski — APPROVED** [review]

File: core/src/test/java/io/questdb/test/griffin/wal/WalAlterTableSqlTest.java (hunk 1 above)
Comment (ideoma) [reply core/src/test/java/io/questdb/test/griffin/wal/WalAlterTableSqlTest.java]: @bziobrowski added the drops, please have a look

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoException.java|2|2|100.00%|

**Review by bziobrowski — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoException.java|2|2|100.00%|

**Pull request merged**